esp32_s3_korvo_2/esp32_s3_korvo_2.c

```diff
@@ -317,7 +317,11 @@ esp_io_expander_handle_t bsp_io_expander_init(void)
     if (io_expander) {
         ESP_LOGD(TAG, "io_expander is initialized");
     } else {
-        BSP_ERROR_CHECK_RETURN_NULL(esp_io_expander_new_i2c_tca9554(BSP_I2C_NUM, BSP_IO_EXPANDER_I2C_ADDRESS, &io_expander));
+        if (esp_io_expander_new_i2c_tca9554(BSP_I2C_NUM, BSP_IO_EXPANDER_I2C_ADDRESS, &io_expander) != ESP_OK) {
```

Comment:
What about try  this `ESP_IO_EXPANDER_I2C_TCA9554A_ADDRESS_000` and when failed, put the newer `ESP_IO_EXPANDER_I2C_TCA9554_ADDRESS_000` and try it again? Then it should work both boards.

Reply:
But our mass-produced version uses TCA9554. If used that way, it means most users will have this error. I think this is not a long-term solution.

Reply:
And what about the first put the newer address and if failed, try the old one and after that show the right error?

Reply:
Sure, that's a good idea! Let me try.

Reply:
@espzav All done!

Reply:
@Lzw655 Thank you very much! LGTM, I will test it on HW and then approve it.